Jenkinsfile

```diff
@@ -0,0 +1,208 @@
+    booleanParam(name: 'intake_ci', defaultValue: true, description: 'Enable test')
+    booleanParam(name: 'test_ci', defaultValue: true, description: 'Enable test')
+    booleanParam(name: 'release_ci', defaultValue: true, description: 'Enable build the release packages')
+    booleanParam(name: 'its_ci', defaultValue: true, description: 'Enable async ITs')
+    string(name: 'DIAGNOSTIC_INTERVAL', defaultValue: "0", description: 'Elasticsearch detailed logging every X seconds')
+    string(name: 'ES_LOG_LEVEL', defaultValue: "error", description: 'Elasticsearch error level')
+  }
+  stages {
+    /**
+    Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout') {
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+      }
+      options { skipDefaultCheckout() }
+      steps {
+        // pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true,
+                    shallow: false, reference: "/var/lib/jenkins/.git-references/${REPO}.git")
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    /**
+    Updating generated files for Beat.
+    Checks the GO environment.
+    Checks the Python environment.
+    Checks YAML files are generated.
+    Validate that all updates were committed.
+    */
+    stage('Intake') {
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+      }
+      when {
+        beforeAgent true
+        allOf {
+          expression { return params.intake_ci }
+          // expression { return env.ONLY_DOCS == "false" }
+        }
+      }
+      steps {
+        // withGithubNotify(context: 'Intake') {
+          deleteDir()
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            withGoEnv(){
+              sh(label: 'Run intake', script: './.ci/scripts/intake.sh')
+            }
+          }
+        }
+      // }
+    }
+    stage('Build and Test'){
+      failFast false
+      parallel {
+        /**
+        Build on a linux environment.
+        */
+        stage('linux build') {
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            allOf {
+              expression { return params.linux_ci }
+            }
+          }
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir(BASE_DIR){
+              withMageEnv(){
+                retry(2) { // Retry in case there are any errors to avoid temporary glitches
+                  sleep randomNumber(min: 5, max: 10)
+                  sh(label: 'Linux build', script: './.ci/scripts/build.sh')
+                }
+              }
+            }
+          }
+        }
+        /**
+          Run unit tests and report junit results.
```

Comment:
since we're running unit tests in the PRs CI we should consider removing one of them, since this is meant to run only post-merge I think we should remove this one as it's redundant, wdyt?

Reply:
I agree, set default to false-these will be skipped